openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1655,6 +1655,7 @@ static int cert_requested(SSL* ssl, X509** x509Out, EVP_PKEY** pkeyOut) {
             break;
     }
 #else
+    (void) ssl2_ctype;
```

Comment:
I think we should be able to declare variables at the beginning of any new scope.

Option 1. Wrap existing declaration in `#if`

```c
tcn_ssl_ctxt_t *c = SSL_get_app_data2(ssl);
#if !defined(OPENSSL_IS_BORINGSSL) && (OPENSSL_VERSION_NUMBER < 0x1000200fL || LIBRESSL_VERSION_NUMBER < 0x20400000L)
char ssl2_ctype = SSL3_CT_RSA_SIGN;
#endif
int ctype_num;
...
```

Option 2. Declare a new scope in the existing `#if` (I haven't tried this so please confirm)

```c
#if !defined(OPENSSL_IS_BORINGSSL) && (OPENSSL_VERSION_NUMBER < 0x1000200fL || LIBRESSL_VERSION_NUMBER < 0x20400000L)
{
     char ssl2_ctype = SSL3_CT_RSA_SIGN;
     switch (ssl->version) {
      ...
}
#else
```